From 6cb9718344816ddda6b189d14550d1c0e00018e6 Mon Sep 17 00:00:00 2001
From: Zdenek Dohnal <zdohnal@redhat.com>
Date: Wed, 7 Feb 2024 14:43:07 +0100
Subject: [PATCH] Fix the daemon crash when Get-Printer-Attributes fails

cups-browsed crashed when it found remote CUPS queue shared by mDNS on
local network, but IPP request for this queue failed. The empty prattrs
is later accessed, which causes the crash.

Fixed by jumping to the point where failed attempts ends when we don't
get the IPP response.
---
 daemon/cups-browsed.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/daemon/cups-browsed.c b/daemon/cups-browsed.c
index 29fd34a1..a633b79c 100644
--- a/daemon/cups-browsed.c
+++ b/daemon/cups-browsed.c
@@ -7815,8 +7815,11 @@ create_remote_printer_entry (const char *queue_name,
       p->prattrs = cfGetPrinterAttributes(p->uri, NULL, 0, NULL, 0, 1);
       debug_log_out(cf_get_printer_attributes_log);
       if (p->prattrs == NULL)
+      {
 	debug_printf("get-printer-attributes IPP call failed on printer %s (%s).\n",
 		     p->queue_name, p->uri);
+	goto fail;
+      }
     }
   }
   else
